browser/extensions/api/brave_rewards_api.cc

```diff
@@ -1230,5 +1236,75 @@ BraveRewardsSaveOnboardingResultFunction::Run() {
   return RespondNow(NoArguments());
 }
 
+BraveRewardsGetPrefsFunction::~BraveRewardsGetPrefsFunction() = default;
+
+ExtensionFunction::ResponseAction BraveRewardsGetPrefsFunction::Run() {
+  auto* rewards_service = RewardsServiceFactory::GetForProfile(
+      Profile::FromBrowserContext(browser_context()));
+
+  if (!rewards_service)
+    return RespondNow(Error("Rewards service is not initialized"));
+
+  rewards_service->GetAutoContributeProperties(base::BindRepeating(
+      &BraveRewardsGetPrefsFunction::GetAutoContributePropertiesCallback,
+      this));
+
+  return RespondLater();
+}
+
+void BraveRewardsGetPrefsFunction::GetAutoContributePropertiesCallback(
+    ledger::type::AutoContributePropertiesPtr properties) {
+  base::Value prefs(base::Value::Type::DICTIONARY);
+  prefs.SetBoolKey("autoContributeEnabled", properties->enabled_contribute);
+  prefs.SetDoubleKey("autoContributeAmount", properties->amount);
+
+  auto* ads_service = AdsServiceFactory::GetForProfile(
+      Profile::FromBrowserContext(browser_context()));
+
+  if (ads_service) {
+    prefs.SetBoolKey("adsEnabled", ads_service->IsEnabled());
+    prefs.SetIntKey("adsPerHour",
+                    static_cast<int>(ads_service->GetAdsPerHour()));
+  } else {
+    prefs.SetBoolKey("adsEnabled", false);
+    prefs.SetDoubleKey("adsPerHour", 0);
+  }
```

Comment:
In one clause we're setting `adsPerHour` as an integer, but in the other we're setting it as a double. Was that intentional?

Reply:
Not intentional, thanks for the catch!